src/crafting.cpp

```diff
@@ -1072,6 +1084,16 @@ void player::complete_craft( item &craft, const tripoint &loc )
     bool first = true;
     size_t newit_counter = 0;
     for( item &newit : newits ) {
+
+        // Points to newit unless newit is a non-empty container, then it points to newit's contents.
+        // Necessary for things like canning soup; sometimes we want to operate on the soup, not the can.
+        item *content_ptr;
+        if( newit.is_container() && !newit.contents.empty() ) {
+            content_ptr = &newit.contents.back();
+        } else {
+            content_ptr = &newit;
+        }
```

Comment:
You can replace this with:
item &food_item = ( newit.is_container() && !newit.contents.empty() ) ? newit.contents.back() : newit;

The pointer isn't *bad* per se, but that does make things tidier, also a more descriptive name is nice.

Reply:
Oh cool, didn't think about trying an inline if statement to satisfy the "must initialize reference" requirement. Will change that now.